core/server/middleware/middleware.js

```diff
@@ -76,6 +81,16 @@ function sslForbiddenOrRedirect(opt) {
     return response;
 }
 
+function verifySessionHash(hash) {
+    var bcryptCompare = Promise.promisify(bcrypt.compare);
```

Comment:
I'd recommend moving the reference to the promisified method up to the top of the file with all the other module level declarations.  There's no need to re-create it on every invocation of `verifySessionHash`.